Modules/HR/Http/Controllers/Recruitment/ApplicationController.php

```diff
@@ -68,9 +68,32 @@ public function index()
         $applications = Application::join('hr_application_round', function ($join) {
             $join->on('hr_application_round.hr_application_id', '=', 'hr_applications.id')
                 ->where('hr_application_round.is_latest', true);
-        })
-            ->with(['applicant', 'job', 'tags', 'latestApplicationRound'])
-            ->whereHas('latestApplicationRound')
+        })->with(['applicant', 'job', 'tags', 'latestApplicationRound']);
+        foreach (array_keys(request()->all()) as $filterKeys) {
+            switch ($filterKeys) {
+                case 'start-year':
+                    $startYear = request()->all()['start-year'] ? (int) request()->all()['start-year'] : null;
+                    if ($startYear != null) {
+                        $applications = $applications->whereHas('applicant', function ($query) use ($startYear) {
+                            $query->where('graduation_year', '>=', $startYear);
+                        });
+                    }
+                    break;
+                case 'end-year':
+                    $endYear = request()->all()['end-year'] ? (int) request()->all()['end-year'] : null;
```

Comment:
I think this should return null if not present
```suggestion
                    $endYear = request()->get('end-year');
```

Reply:
Please check and update at other places